cfme/tests/infrastructure/test_provisioning_rest.py

```diff
@@ -111,6 +111,43 @@ def test_provision(request, appliance, provider, provision_data):
     assert found_vms, 'VM `{}` not found'.format(vm_name)
 
 
+@pytest.mark.rhv2
+@pytest.mark.uncollectif(lambda provider: not provider.one_of(RHEVMProvider),
+                         reason='These profile names are RHV specific.')
+@pytest.mark.parametrize('vnic_profile', ['no_vnic_profile', 'specific_vnic_profile'])
+def test_provision_vlan(request, appliance, provision_data, vnic_profile, provider):
+    """Tests provision via REST API.
+    Prerequisities:
+        * Have a provider set up with templates suitable for provisioning.
+    Steps:
+        * POST /api/provision_requests (method ``create``) the JSON with provisioning data. The
+            request is returned.
+        * Apart from the usual provisioning settings, set vlan
+          to values <Empty>/profile_name (network_name)
+        * Query the request by its id until the state turns to ``finished`` or ``provisioned``.
+    Metadata:
+        test_flag: rest, provision
+    """
+    vm_name = provision_data['vm_fields']['vm_name']
+    if vnic_profile == 'no_vnic_profile':
+        vnic_profile = '<Empty>'
+    elif vnic_profile == 'specific_vnic_profile':
+        vlan_data = provider.data["provisioning"]["vlan"]
+        vnic_profile = '{0} ({0})'.format(vlan_data)
+    provision_data['vm_fields']['vlan'] = vnic_profile
+    request.addfinalizer(lambda: clean_vm(appliance, provider, vm_name))
+    appliance.rest_api.collections.provision_requests.action.create(**provision_data)
+    assert_response(appliance)
+    provision_request = appliance.collections.requests.instantiate(description=vm_name,
+                                                                   partial_check=True)
+    provision_request.wait_for_request()
+    msg = "Provisioning failed with the message {}".format(
+        provision_request.rest.message)
+    assert provision_request.is_succeeded(), msg
+    found_vms = appliance.rest_api.collections.vms.find_by(name=vm_name)
+    assert found_vms, 'VM `{}` not found'.format(vm_name)
```

Comment:
Here we are checking if the VM has been provisioned. While that's definitely an important check, I believe check of the new VM's actual NICs is in place here, since this test case is focused on this area. You could do something like this:
`vm = appliance.collections.infra_vms.instantiate(vm_name, provider)`
and then you have this at your disposal:
`vm.mgmt.get_nics()`
By these means you can check that the VM actually (or does not have) the expcted NIC configuration.

I am saying this because I was running this specific test case with `vnic_profile == 'no_vnic_profile'` and the provisioned VM had a NIC like [this](https://up1-vrutkovs.int.open.paas.redhat.com/#rFIK31aKn14oSAw-7Xo9_Q) even though that the used template has not had any NICs. This might be a product bug. But regardless of whether this is a bug or not, I believe it would be good to have such a check here.

Reply:
@jan-zmeskal I think the problem here is in naming. The `vnic_profile == 'no_vnic_profile'` you mention actually refers to Empty profile (on line 133). Although this profile is called Empty (which I believe means no QoS limitation or qota), it is a profile. Hence there is a NIC. If you create a VM from a template and do not select any profile (not even the Empty one), the created VM won't have a NIC. This is a desired behaviour AFAIK.

So I would suggest renaming the `no_vnic_profile` to `empty_vnic_profile`. Additionally, `no_vnic_profile` may be tested by something  like: `vnic_profile = None` or `vnic_profile = '<Use template nics>'` (where both options are effectively the same, so I would prefer the second one for better readability).

Reply:
@rhrazdil Thank you very much for clarification. So it means that the behaviour I observed is not a product bug, which is good. However I still think my main point is valid - IMO it would be reasonable to check presence/absence of the desired NIC(s) via wrapanapi in the final part of the test.

Reply:
@jan-zmeskal Yes, I agree. By the way, I played around a little with provisioning with different vnic profile configurations and I discovered that if you select in CFME provisioning dialog option 'No profile', the provisioned VM does have a NIC with '<Empty>' profile. I'll file a bug for this naming discrepancy because I think I'm not the only one who may find it confusing.

Reply:
@rhrazdil You definitely are not - I got lost in this as well.

Reply:
* Regarding the "<Empty>" option:
This is from https://bugzilla.redhat.com/show_bug.cgi?id=1449157 doc text:

3. Provision a virtual machine with one vNic where the vNic has no profile.

"vm_fields" : {
    ...
    "vlan" : "<Empty>"
  }

The history was that in the UI we had option "<Empty>" added, and the corresponding phrase in REST api was also "Empty".
There were complains on the misleading naming in the UI,  so the UI changed from <Empty> into <No Profile>:
https://github.com/ManageIQ/manageiq-providers-ovirt/pull/151

However in REST api, the "<Empty>" still remained.

* Regarding the test name: I shall rename the test name from "no_vnic_profile" into "empty_vnic_profile"

* Regarding adding the test a check of the actual vnic profile set - I'll need to work on this further - so moving the PR back to [WIP].